src/blueprint/upstream.rs

```diff
@@ -42,14 +42,23 @@ impl Upstream {
 impl Default for Upstream {
     fn default() -> Self {
         // NOTE: Using unwrap because try_from default will never fail
-        Upstream::try_from(config::Upstream::default()).unwrap()
+        Upstream::try_from(&ConfigModule::default()).unwrap()
     }
 }
 
-impl TryFrom<crate::config::Upstream> for Upstream {
+impl TryFrom<&ConfigModule> for Upstream {
     type Error = ValidationError<String>;
 
-    fn try_from(config_upstream: config::Upstream) -> Result<Self, Self::Error> {
+    fn try_from(config_module: &ConfigModule) -> Result<Self, Self::Error> {
+        let config_upstream = config_module.upstream.clone();
+
+        let mut allowed_headers = config_upstream.get_allowed_headers();
+
+        if config_module.extensions.has_auth() {
+            // force add auth specific headers to use it to make actual validation
+            allowed_headers.insert(hyper::header::AUTHORIZATION.to_string());
```

Comment:
Update the documentation with this new information. Currently, allowed headers defaults to being empty. What if someone wants to not send this header? Or wants to send a different header in future? Something to think about.